### PR TITLE
Changed <amp-analytics> tag to read config from a script child.

### LINKED ISSUE #1081: [Amp Analytics]: Move analytics config inside a child script tag of <amp-analytics>
```html
<amp-analytics id="analytics1">
  <script type="application/json">
   {
    "host": "my-analytics.com",
    ...
   }
  </script>
</amp-analytics>
```

Discussion here: https://github.com/ampproject/amphtml/pull/982#discussion_r46492360

### PR DESCRIPTION
**Pull request by avimehta**
This commit fixes #1081 and is related to #871.

### CODE REVIEW

**Comment by dvoytenko** [comment]
Ready for review?

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 1]
```diff
@@ -126,10 +126,17 @@ export class AmpAnalytics extends AMP.BaseElement {
...
+      if (children.length == 1 && children[0].tagName.toUpperCase() == 'SCRIPT') {
+        inline = JSON.parse(children[0].textContent);
+      } else if (children.length > 1) {
+        log.warn(this.getName_(),
+            'The tag should contain only one <script> child.');
+      }
     }
     catch (er) {
-      log.warn(this.getName(), "Analytics config could not be parsed.");
+      log.warn(this.getName(), 'Analytics config could not be parsed. ' +
```
Comment (dvoytenko) [inline extensions/amp-analytics/0.1/amp-analytics.js]: `log.warn(TAG_, 'ABC', er)`

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 2]
```diff
@@ -126,10 +126,17 @@ export class AmpAnalytics extends AMP.BaseElement {
 
     let inline = {};
     try {
-      inline = JSON.parse(this.element.textContent);
+      const children = this.element.children;
+      if (children.length == 1 && children[0].tagName.toUpperCase() == 'SCRIPT') {
```
Comment (dvoytenko) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Please test `type` as well to be "application/json"

File: extensions/amp-analytics/0.1/test/test-amp-analytics.js [hunk 3]
```diff
@@ -49,7 +49,9 @@ describe('amp-analytics', function() {
   function getAnalyticsTag(config, attrs) {
     config = JSON.stringify(config);
     const el = document.createElement('amp-analytics');
-    el.textContent = config;
+    const script = document.createElement('script');
```
Comment (dvoytenko) [inline extensions/amp-analytics/0.1/test/test-amp-analytics.js]: Definitely `type` attribute here

**Comment by dvoytenko** [comment]
Also, see travis errors.

File: extensions/amp-analytics/0.1/test/test-amp-analytics.js (hunk 3 above)
Comment (avimehta) [inline extensions/amp-analytics/0.1/test/test-amp-analytics.js]: done.

File: extensions/amp-analytics/0.1/amp-analytics.js (hunk 2 above)
Comment (avimehta) [inline extensions/amp-analytics/0.1/amp-analytics.js]: done.

File: extensions/amp-analytics/0.1/amp-analytics.js (hunk 1 above)
Comment (avimehta) [inline extensions/amp-analytics/0.1/amp-analytics.js]: done.

**Comment by dvoytenko** [comment]
LGTM

**Comment by avimehta** [comment]
@cory-work: I did not run the validation with the commit. I didn't realize that the analytics validation was already in place. I'll look into this. 

cc: @btownsend

**Pull request merged**